nntrainer/dataset/data_producers.h

```diff
@@ -43,12 +44,39 @@ class DataProducer {
 
   /**
    * @brief create an iteration
+   * @todo rename this to BatchGenerator
    * @return Iteration iteration, if std::get<0>(retval) == true means end of
    * iteration, at the end of the iteration, it's responsibility of @a this to
    * shuffle.
    */
   using Generator = std::function<Iteration(void)>;
 
+  /**
+   * @brief Sample represents a view of single element which can be fed to the
+   * model. It is the smallest unit to produce a data
+   * @return std::get<0>(Sample) denotes inputs
```

Comment:
@retval?

Reply:
This is not actually a return value, but explanation about the type. I have revised this to not to be a tuple in the later PR :)